### ISSUE TITLE
No image with electron 0.36

### ISSUE THREAD

**Issue by MikeGCV** [RSATom/WebChimera.js#59]
Hello, i've  successfully build webchimera.js against electron 0.36.0, but i can't get no image with wcjs-renderer.

This was in windows x64

On 0.35.4, no problem

Thank you

**Comment by RSATom** [comment]
Thanks, I'll check soon.

**Comment by RSATom** [comment]
Checked Electron v0.36 with https://github.com/RSATom/wcjs-ugly-demo - it works.
Could you please try it too?

**Comment by MikeGCV** [comment]
I will try it and retry in my application. I forgot to refer that i'am using webchimera inside a webview, i don't know if they changed something in 0.36.0.. I will post my results later.
Thank you

**Comment by MikeGCV** [comment]
Just tried with https://github.com/RSATom/wcjs-ugly-demo, no image, no sound, really strange.
Windows 10 and 8.1
Does this need new vlc plugins?

**Comment by RSATom** [comment]
Tried on win10 pro x64 with VLC 2.2.1 
What VLC version do you use?

**Comment by MikeGCV** [comment]
Same version here, really strange

**Comment by RSATom** [comment]
maybe you could see something in DebugView?

**Comment by MikeGCV** [comment]
The files that i copy to the Release folder are:
libvlccore.dll
libvlc.dll
and the plugins folder

DebugView, how can i activate that?

Thanks

**Comment by RSATom** [comment]
https://technet.microsoft.com/en-us/sysinternals/debugview.aspx
Just run it before Electron

**Comment by MikeGCV** [comment]
"[1623900] core libvlc error: No plugins found! Check your VLC installation."

Strange the plugins are in the same folder

**Comment by RSATom** [comment]
it's possible it missing some additional dlls. Try add full VLC installation to PATH instead of copy.

**Comment by RSATom** [comment]
it's strange, but I can't get it working on another PC with Electron v0.36. Trying to find the reason...

**Comment by MikeGCV** [comment]
Just builded again wcjs-ugly-demo and added the full installation of VLC to the following folder:

wcjs-ugly-demo\node_modules\webchimera.js\build\Release

Same error

**Comment by RSATom** [comment]
hm... can you try run `vlc.exe` from wcjs-ugly-demo\node_modules\webchimera.js\build\Release? Will it work?

**Comment by RSATom** [comment]
found why it didn't work - just tried run x64 `Electron` with ia32 `WebChimera.js.node`. After install correct Electron - all works fine.

**Comment by RSATom** [comment]
I have one else idea, could you please check, if you have `libvlc.dll` in your windows\system32 folder or somewhere else in Windows system folders?

**Comment by MikeGCV** [comment]
vlc works from that folder, i don't have libvlc.dll anywhere else, just rebuilded using 0.35.4 in wcjs-ugly-demo (x64) and all works fine

**Comment by RSATom** [comment]
hm... I've tested with 32 bit image. Maybe it's x64 VLC issue. Could you please try it with Electron ia32?

**Comment by RSATom** [comment]
Checked with Electron x64 + wcjs x64 + VLC 2.2.1 x64 - all works fine...

**Comment by MikeGCV** [comment]
I will try with ia32, i don't understand what i'm doing wrong

**Comment by RSATom** [comment]
I think the most possible thing - you mix x64 and ia32 modules, but I can't be sure.

**Comment by MikeGCV** [comment]
I will retry today on windows and linux, Post results later

**Comment by jaruba** [comment]
I ran into this issue too, Electron, wcjs, VLC all 64bit.. same error in DebugView.

**Comment by jaruba** [comment]
I also tried setting the vlc path in the env vars, still getting the same error.

**Comment by jaruba** [comment]
I tried electron 0.36.1 too, same issue there also..

**Comment by RSATom** [comment]
@jaruba, I have one idea about reason, will check it asap

**Comment by jaruba** [comment]
@RSATom want me to make some testing environment and send it to you?

**Comment by RSATom** [comment]
you could download [Process Monitor](https://technet.microsoft.com/en-US/sysinternals/processmonitor.aspx) and try understand how to monitor file access with it

**Comment by RSATom** [comment]
btw, where do you have Electron installed? global or in project's `node_modules`?

**Comment by jaruba** [comment]
using `electron-prebuilt`, so `/node_modules/electron-prebuilt/dist/`

**Comment by RSATom** [comment]
ha.. got the same error!

**Comment by jaruba** [comment]
awesome :)

keep in mind, everything works fine in 0.35.x with `electron-prebuilt`, so something was changed in 0.36.x :(

**Comment by RSATom** [comment]
I've found the reason. It's complex issue involving how Electron loads .node modules, how libvlc searches plugins, and how Windows search dependant dlls...

Trying to find solution...

**Comment by jaruba** [comment]
if you can't find an easy enough solution, maybe the proper approach would be to contact electron in hopes that they can revert whatever they did to break it.

**Comment by RSATom** [comment]
tbh I think it's a bug in libvlc. It doesn't support so called UNC Paths

**Comment by RSATom** [comment]
https://trac.videolan.org/vlc/ticket/16187#ticket

still trying to find workaround...

**Comment by MikeGCV** [comment]
Ok, same results, i see that @jaruba is having the same issues, i was afraid that i was doing something wrong and wasting your time @RSATom.

Thank you for checking the issue

**Comment by RSATom** [comment]
@MikeGCV
first of all, as I found this issue related to both `x64` and `ia32` builds
Possible workarounds:
a) place path to `libvlc.dll` to PATH env variable (but you **should not** have `libvlc.dll` near the `WebChimera.js.node` then)
b) set `VLC_PLUGIN_PATH` environment variable before `Electron` launch. For example inside `.bat`/`.cmd` file.

**Comment by RSATom** [comment]
And as conclusion, I still thinking it's not `Electron` or `WebChimera.js` bug, but `libvlc` bug.

**Comment by MikeGCV** [comment]
Hello,
in fact, setting VLC_PLUGIN_PATH before electron solves the issue

**Comment by RSATom** [comment]
Thank you! It seems we need place it to some sort of FAQ...

**Comment by RSATom** [comment]
https://github.com/RSATom/WebChimera.js/wiki/Electron-v0.36.x-compatibility-issue-on-Windows

closed since it's `libvlc` issue

**Comment by sunabozu** [comment]
I can confirm both the issue and the solution. Before loading the window I just do the following:

```javascript
if(process.platform == 'win32')
	process.env['VLC_PLUGIN_PATH'] = path.join(__dirname, 'node_modules/webchimera.js/plugins')
```

**Comment by guidupuy** [comment]
Not working for me... Would anyone care to help? I'm using wcjs-prebuilt with Electron 0.37.8.

Here's what I have in my `main.js` file, in the very first lines before any window is opened:

``` JavaScript
if(process.platform == "win32"){
    process.env["VLC_PLUGIN_PATH"] = path.join(global.appPath, "node_modules/wcjs-prebuilt/bin/plugins");
}
  ```

Also, here's where my VLC files reside:
```
app\node_modules\wcjs-prebuilt\bin
├── plugins
├── index.js
├── libvlc.dll
├── libvlccore.dll
└── WebChimera.js.node
```

**Comment by jaruba** [comment]
@guidupuy does the sample from [this comment](https://github.com/jaruba/wcjs-player/issues/42#issuecomment-196653547) work for you?

**Comment by RSATom** [comment]
@guidupuy, please look debug log with https://technet.microsoft.com/en-us/sysinternals/debugview.aspx

**Comment by guidupuy** [comment]
@RSATom I'm not seeing anything in the logs - here's my Capture "config":

![capture d ecran 2016-05-13 a 09 12 09](https://cloud.githubusercontent.com/assets/1026103/15240567/f0d759ac-18ea-11e6-9bfb-d3cd69c999a1.png)

I can't capture kernel right now because:

![capture d ecran 2016-05-13 a 09 13 45](https://cloud.githubusercontent.com/assets/1026103/15240588/094c618a-18eb-11e6-99d7-5e1ec544682c.png)

I'm pretty sure that's not needed though... Can you confirm?

**Comment by guidupuy** [comment]
@jaruba I'd love to be able to have a working `x64` version - of course if that's not possible I'll switch to `ia32`. For the record, has anyone been able to get WebChimera `x64` to work with Electron `x64`?

**Comment by RSATom** [comment]
> I'm pretty sure that's not needed though... Can you confirm?
Yes, you don't need "Capture Kernel"

Is there any chance you have running Visual Studio at the same time?

**Comment by RSATom** [comment]
>  For the record, has anyone been able to get WebChimera  x64  to work with Electron  x64 ?
Yes, I've tried it and it worded for me.

**Comment by guidupuy** [comment]
I'll find a way, though I'm actually not sure the problem is with the path: if I open the Electron console and require `wcjs-prebuilt`, I get `Uncaught Error: %1 is not a valid Win32 application.` That's weird because I'm running Electron `x64` with VLC `x64`...

**Comment by RSATom** [comment]
this means you try load 32 bit `WebChimera.js.node`

**Comment by guidupuy** [comment]
Doing that now - will let you know how it goes.

**Comment by guidupuy** [comment]
I'm getting the same error with 32-bit `WebChimera.js.node`. Weird huh? Here's the reference of the release, FYI: `WebChimera.js_v0.2.4_electron_v0.37.8_VLC_v2.2.2_ia32_win.zip`

**Comment by guidupuy** [comment]
Wait... Looking at my console logs it seems I may have been trying with `ia32` all along when I thought I had `x64`... Ouch. Let me try again with `x64`.

**Comment by guidupuy** [comment]
Alright so: not getting the `Uncaught Error: %1 is not a valid Win32 application` error anymore, and my webview's not crashing anymore. I now have logs, and I think this will be an easy fix - apparently my MRL isn't well-formed:

![capture d ecran 2016-05-13 a 09 39 56](https://cloud.githubusercontent.com/assets/1026103/15241205/c371dcf4-18ee-11e6-9dcd-76d512466663.png)

**Comment by RSATom** [comment]
yes, you should use `file:///` prefix

**Comment by guidupuy** [comment]
Damn... Silly me. Ok, fixed now. Thanks!
I'm having serious performance problems though. I'll have to see what it's due to.

**Comment by guidupuy** [comment]
Actually, it's not performance problems so much as the following: I'm loading an iframe on top of the canvas on which WebChimera paints the frames. As soon as the iframe shows, the video freezes...

**Comment by guidupuy** [comment]
Happens on Windows only - not on OSX

**Comment by guidupuy** [comment]
I can probably find a workaround: if I set `visibility: hidden` on the iframe, the freeze doesn't happen. Even better: if I manually (eg through the console) show the iframe after it's been loaded, everything works. Probably not worth trying to solve...

**Comment by guidupuy** [comment]
Actually, the problem was with `wcjs-renderer`. I've submitted a PR there.
